### PR TITLE
Keep the drone close enough to the user to keep it visible for the user

### PR DESCRIPTION
**Pull request by Anghul**

### CODE REVIEW

**Review by jonasblanc — APPROVED** [review]
Good code

File: app/src/main/java/ch/epfl/sdp/drone3d/ui/map/MissionInProgressActivity.kt [hunk 1]
```diff
@@ -298,6 +299,32 @@ class MissionInProgressActivity : BaseMapActivity() {
...
+            it?.let {
+                if (locationService.isLocationEnabled() && locationService.getCurrentLocation() != null) {
+                    val distanceUser: Double = it.distanceTo(locationService.getCurrentLocation()!!)
+                    val maxDistance: Double = if (MAX_DIST_TO_USER > weatherReport.value!!.visibility) {
+                        weatherReport.value!!.visibility.toDouble()
+                    } else {
+                        MAX_DIST_TO_USER
+                    }
+
+                    if (distanceUser > maxDistance && !droneData.isMissionPaused().value!!) {
```
Comment (loh0um) [inline app/src/main/java/ch/epfl/sdp/drone3d/ui/map/MissionInProgressActivity.kt:315]: Is it really safe to use value!! here without any check for null values?

**Review by loh0um — APPROVED** [review]

File: .codeclimate.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ checks:
       threshold: 40
   method-complexity:
     config:
-      threshold: 25
+      threshold: 35
```
[the remark this answers is not in the document]
Comment (Anghul) [reply .codeclimate.yml:16]: I didn't find a way to do this here and in an other function in an other branch

File: app/src/main/java/ch/epfl/sdp/drone3d/ui/map/MissionInProgressActivity.kt (hunk 1 above)
Comment (Anghul) [reply app/src/main/java/ch/epfl/sdp/drone3d/ui/map/MissionInProgressActivity.kt:315]: When droneData is set when a simulation is connected, all the MutableLiveData are set to a default value, so it shouldn't be a problem

**Pull request merged**